controller/src/main/java/io/pravega/controller/store/stream/ZKStreamMetadataStore.java

```diff
@@ -460,6 +463,35 @@ public void unregisterBucketListener(int bucket) {
                           });
     }
 
+    /**
+     * When managing the Controller's metadata in Zookeeper, we explicitly create parent bucket zNodes (so they are of
+     * type "zNode"). Otherwise, they may be inadvertently created as Zookeeper "containers" by Curator. This would lead
+     * these zNodes to be candidates for automatic removal by Zookeeper if they become empty.
+     */
+    @Override
+    public void initializeMetadataStore() {
+        List<CompletableFuture<Void>> initializationFutures = new ArrayList<>();
+        Retry.RetryWithBackoff retryPolicy = Retry.withExpBackoff(100, 2, 10);
+        Predicate<Throwable> isDataExistsException = ex -> Exceptions.unwrap(ex) instanceof StoreException.DataExistsException;
+        for (int bucket = 0; bucket < bucketCount; bucket++) {
+            final String bucketPath = String.format(BUCKET_PATH, bucket);
+            initializationFutures.add(retryPolicy.retryWhen(isDataExistsException.negate()).run(() ->
```

Comment:
Does curator have its own retry mechanism? Do we need to retry ourselves?

Reply:
Done.